apple/internal/templates/macos.template.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+# Copyright 2015 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eu
+
+if [[ "$(uname)" != Darwin ]]; then
+  echo "Cannot run macOS targets on a non-mac machine."
+  exit 1
+fi
+
+readonly TEMP_DIR=$(mktemp -d "${TMPDIR:-/tmp}/bazel_temp.XXXXXX")
+
+trap 'rm -rf "${TEMP_DIR}"' ERR EXIT
+
+readonly APP_DIR="${TEMP_DIR}/extracted_app"
+mkdir "${APP_DIR}"
+
+if [ -d '%app_path%' ]; then
+  # App bundles are directories with the .app extension
+  mv '%app_path%' "${APP_DIR}"
```

Comment:
is it needed to move the app into a different location? how about:

```
if -d app_path
   APP_DIR=app_path
else:
   tempdir = mktemp bazel_temp.xxx
   unzip -qq app_path -d temp_dir
   APP_DIR=tempdir/app_name.app

bundle_info_plist = app_dir/Contents/Info.plist
bundle_executable=...

app_dir/contents/MacOS/bundle_executable
```

this way, a single directory is created (besides the unzip) and only if the app was zipped

Reply:
Sounds good. Nope this was just a quick thing to do when I was adding support for tree artifacts but I agree it’s unnecessary

Reply:
@sergiocampama do you think that checking the extension of `app_path` is required here or is checking that it's a directory sufficient for now?